File: PureMPA/src/main/java/com/example/simple_todo/domain/Task.java
package com.example.simple_todo.domain;

public record Task(TaskId id, String description, boolean isCompleted) {

    public Task(TaskId id, String description) {
        this(id, description, false);
    }

    public Task complete() {
        return new Task(
                this.id,
                this.description,
                true
        );
    }
}

File: ReactSPA/simple-todo/src/types/Task.ts
type TaskId = number;

interface Task {
  readonly id: TaskId;
  description: string;
  readonly isCompleted: boolean;
}

function completeTask(task: Task): Task {
  return {
    id: task.id,
    description: task.description,
    isCompleted: true
  }
}

export type { TaskId, Task };
export { completeTask };
export default Task;